conda/core/path_actions.py

```diff
@@ -1042,3 +1042,34 @@ def target_full_path(self):
     def __str__(self):
         return ('ExtractPackageAction<source_full_path=%r, target_full_path=%r>'
                 % (self.source_full_path, self.target_full_path))
+
+
+class DumpRecordAction(PathAction):
+
+    def __init__(self, target_pkgs_dir, target_extracted_dirname, record):
+        self.target_pkgs_dir = target_pkgs_dir
+        self.target_extracted_dirname = target_extracted_dirname
+        self.record = record
+
+    def verify(self):
+        self._verified = True
+
+    def execute(self):
+
+        meta = join(self.target_full_path, 'info', 'record.json')
+        log.trace("writing record to %s", meta)
+        with open(meta, 'w') as out:
+            out.write(self.record.pretty_json())
```

Comment:
Awe, I don't think I added `sort=True` by default.  We could pass it through the kwargs.  Or you could also make the line

    out.write(json.dumps(record, indent=2, separators=(',', ': '), sort_keys=true, cls=EntityEncoder))

or

    out.write(json.dumps(record.dump(), indent=2, separators=(',', ': '), sort_keys=true))

Mostly this comment is just FYI for different ways to work with the Entity object.  The `.dump()` method strips all entity information and just gives back a plain old python dictionary.